### PR TITLE
 feat: 🎸 trusted claim issuers: set, add, remove

### PR DESCRIPTION
**Pull request by sansan**
### JIRA Link 
https://polymesh.atlassian.net/browse/DA-61

### Changelog / Description 

- moves list trusted claim issuers to a separate controller
- Adds endpoints to add, set and remove trusted claim issuers

### Checklist - 

- [x] New Feature ?
- [x] Updated swagger annotation (if API structure is changed) ?
- [x] Unit Test (if possible) ?
- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/common/decorators/swagger.ts [hunk 1]
```diff
@@ -126,11 +126,15 @@ type SupportedHttpStatusCodes =
  * @param messages - key value map of HTTP response code to their description that will be passed to appropriate `MethodDecorator`
  */
 export function ApiTransactionFailedResponse(
-  messages: Partial<Record<SupportedHttpStatusCodes, string>>
+  messages: Partial<Record<SupportedHttpStatusCodes, string | string[]>>
```
Comment (polymath-eric) [inline src/common/decorators/swagger.ts:129]:
```suggestion
  messages: Partial<Record<SupportedHttpStatusCodes, string[]>>
```
It's not a big deal adding in an extra `[]` for a single item and then it's easier to add a second.

File: src/test-utils/service-mocks.ts [hunk 2]
```diff
@@ -180,6 +179,13 @@ export class MockAuthService {
   validateApiKey = jest.fn();
 }
 
+export class MockTrustedClaimIssuersService {
```
Comment (polymath-eric) [inline src/test-utils/service-mocks.ts:182]: Prashant added this package recently: https://www.npmjs.com/package/@golevelup/ts-jest

You should be able to do `createMock<MockTrustedClaimIssuersService>()` where ever you need this instead.

You might want to define a Nest `Provder` object in this file to save on test boiler plate.

**Review by polymath-eric — COMMENTED** [review]
Can you update this to be against `feature-release` (or `master` if it gets merged)?

File: src/common/decorators/swagger.ts (hunk 1 above)
Comment (polymath-eric) [reply src/common/decorators/swagger.ts:129]: `Arrary.isArrary` -> `rawDescription.length > 1` ?

I won't block approval on this though, either way is OK

File: src/common/decorators/swagger.ts [hunk 3]
```diff
@@ -126,11 +126,15 @@ type SupportedHttpStatusCodes =
...
 ): ReturnType<typeof applyDecorators> {
   const decorators: MethodDecorator[] = [];
 
-  Object.entries(messages).forEach(([statusCode, description]) => {
+  Object.entries(messages).forEach(([statusCode, rawDescription]) => {
+    const description = Array.isArray(rawDescription)
+      ? '<ul>' + rawDescription.map(v => `<li>${v}</li>`).join(' ') + '</ul>'
```
Comment (polymath-eric) [inline src/common/decorators/swagger.ts:135]: (style) Usually we don't do single letter variables, even if they are close to the definition.
```suggestion
      ? '<ul>' + rawDescription.map(description => `<li>${description}</li>`).join(' ') + '</ul>'
```

I still find myself doing single letters for things like this from time to time though.

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]
While your at it, the `trustedFor` in `TrustedClaimIssuerDto` needs to be amended to support `null`. Can you take care of that as well?

File: src/compliance/trusted-claim-issuers.controller.spec.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        claimIssuers: [],
+        signer: 'Alice',
+      };
+
+      const response = {
+        transactions: [transaction],
+      };
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```
Comment (polymath-eric) [inline src/compliance/trusted-claim-issuers.controller.spec.ts:125]: why ts ignore?

File: src/compliance/dto/trusted-claim-issuer.dto.ts [hunk 5]
```diff
@@ -10,10 +10,11 @@ export class TrustedClaimIssuerDto {
       'List of Claim types for which an Identity is trusted for verifying. Defaults to all types',
     enum: ClaimType,
     isArray: true,
+    nullable: true,
   })
   @IsOptional()
   @IsEnum(ClaimType, { each: true })
-  readonly trustedFor?: ClaimType[];
+  readonly trustedFor?: ClaimType[] | null;
```
Comment (polymath-eric) [inline src/compliance/dto/trusted-claim-issuer.dto.ts:17]:
```suggestion
  readonly trustedFor: ClaimType[] | null = null;
```
If this gets defaulted to null, then we don't have to deal with the undefined case, which means a type assertion isn't needed in the service.

File: src/compliance/dto/trusted-claim-issuer.dto.ts [hunk 6]
```diff
@@ -10,10 +10,11 @@ export class TrustedClaimIssuerDto {
       'List of Claim types for which an Identity is trusted for verifying. Defaults to all types',
     enum: ClaimType,
     isArray: true,
+    nullable: true,
```
Comment (polymath-eric) [inline src/compliance/dto/trusted-claim-issuer.dto.ts:13]:
```suggestion
    nullable: true,
    default: null
```

File: src/compliance/trusted-claim-issuers.service.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    return asset.compliance.trustedClaimIssuers.get();
+  }
+
+  public async set(ticker: string, params: SetTrustedClaimIssuersDto): ServiceReturn<Asset> {
+    const { signer, webhookUrl, ...rest } = params;
+    const asset = await this.assetsService.findOne(ticker);
+
+    return this.transactionsService.submit(
+      asset.compliance.trustedClaimIssuers.set,
+      rest as ModifyAssetTrustedClaimIssuersAddSetParams,
```
Comment (polymath-eric) [inline src/compliance/trusted-claim-issuers.service.ts:34]:
```suggestion
      rest,
```
If the DTO matches the signature there is no need to assert this.

The SDK is checking explicitly for null in a function, so this assertion is unsound.

`(aTrustedFor === null && bTrustedFor === null)`
https://github.com/PolymathNetwork/polymesh-sdk/blob/339b750301f99ccff0c390a00fdbf1f153765999/src/api/procedures/modifyAssetTrustedClaimIssuers.ts#L84

File: src/test-utils/mocks.ts [hunk 8]
```diff
@@ -6,18 +6,33 @@ import {
...
       : T[K];
   };
 
+export const mockTrustedClaimIssuer = createMock<TrustedClaimIssuer<true>>();
+
+export const createMockTransactionResult = <T>({
+  transactions,
+  result,
+}: {
+  transactions: TransactionResult<any>['transactions'];
```
Comment (polymath-eric) [inline src/test-utils/mocks.ts:30]:
```suggestion
  transactions: TransactionResult<T>['transactions'];
```
?

**Review by polymath-eric — APPROVED** [review]
LGTM

**Review by prashantasdeveloper — APPROVED** [review]

**Pull request merged**